IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -0,0 +1,539 @@
+                                                             error:(__unused NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return nil;
+}
+
+// Remove items with the given Metadata key from the macOS keychain cache.
+- (BOOL)removeItemsWithMetadataKey:(__unused MSIDCacheKey *)key
+                           context:(__unused id<MSIDRequestContext>)context
+                             error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return FALSE;
+}
+
+#pragma mark - Wipe Info
+
+// Saves information about the app which most-recently removed a token.
+- (BOOL)saveWipeInfoWithContext:(__unused id<MSIDRequestContext>)context
+                          error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return FALSE;
+}
+
+// Read information about the app which most-recently removed a token.
+- (NSDictionary *)wipeInfo:(__unused id<MSIDRequestContext>)context
+                     error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return nil;
+}
+
+#pragma mark - clear
+
+// A test-only method that deletes all items from the cache for the given context.
+- (BOOL)clearWithContext:(id<MSIDRequestContext>)context
+                   error:(NSError **)error
+{
+    MSID_TRACE;
+    MSID_LOG_WARN(context, @"Clearing the whole context. This should only be executed in tests");
+
+    // for now, this just deletes all accounts
+    return [self removeItemsWithAccountKey:[MSIDCacheKey new] context:context error:error];
+}
+
+#pragma mark - Utilities
+
+// Get the basic/default keychain query dictionary for account items.
+- (NSMutableDictionary *)defaultAccountQuery:(MSIDCacheKey *)key
+{
+    MSID_TRACE;
+    NSMutableDictionary *query = [NSMutableDictionary new];
+    query[(id)kSecClass] = (id)kSecClassGenericPassword;
+
+    if (key.account.length > 0)
+    {
+        query[(id)kSecAttrAccount] = key.account; // <homeAccountId>-<environment>
+    }
+    if (key.service.length > 0)
+    {
+        query[(id)kSecAttrService] = key.service; // <realm>
+    }
+
+    // Add a marker for our cache items in the keychain.
+    // It avoids keychain errors, in particular with clearWithContext.
+    // This property is a FourCC integer, not a string:
+    query[(id)kSecAttrCreator] = [NSNumber numberWithUnsignedInt:'MSAL'];
+    // Note: Would something like this be better?
+    // query[(id)kSecAttrSecurityDomain] = @"com.microsoft.msalcache";
+
+    return query;
+}
+
+// Get the basic/default keychain update dictionary for account items.
+// These are not _primary_ keys, but if they're present in the key object we
+// want to set them when adding/updating.
+- (NSMutableDictionary *)defaultAccountUpdate:(MSIDCacheKey *)key
+{
+    MSID_TRACE;
+    NSMutableDictionary *update = [NSMutableDictionary new];
+
+    if (key.generic.length > 0)
+    {
+        update[(id)kSecAttrGeneric] = key.generic;
```

Comment:
I don't see this in the comments

Reply:
Comment added. This is used for <authority_account_id>, or account.username in our case.